### PR TITLE
mds: Reorganize structure members in SnapRealm header

### PR DESCRIPTION
**Pull request by varshar16**
Fixes: https://tracker.ceph.com/issues/43386
Signed-off-by: Varsha Rao <varao@redhat.com>

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/mds/SnapRealm.h [hunk 1]
```diff
@@ -153,8 +122,38 @@ struct SnapRealm {
...
   }
+
+  // realm state
+  sr_t srnode;
+
+  // in-memory state
+  MDCache *mdcache;
+  CInode *inode;
+
+  mutable bool open = false;                        // set to true once all past_parents are opened
```
Comment (joscollin) [inline src/mds/SnapRealm.h]: private

File: src/mds/SnapRealm.h [hunk 2]
```diff
@@ -153,8 +122,38 @@ struct SnapRealm {
...
+  bool past_parents_dirty = false;
+  bool global;
```
Comment (joscollin) [inline src/mds/SnapRealm.h]: private

File: src/mds/SnapRealm.h [hunk 3]
```diff
@@ -153,8 +122,38 @@ struct SnapRealm {
...
+
+  SnapRealm *parent;
+  set<SnapRealm*> open_children;    // active children that are currently open
+  set<SnapRealm*> open_past_children;  // past children who has pinned me
+  map<inodeno_t, pair<SnapRealm*, set<snapid_t> > > open_past_parents;  // these are explicitly pinned.
```
Comment (joscollin) [inline src/mds/SnapRealm.h]: private

File: src/mds/SnapRealm.h [hunk 4]
```diff
@@ -153,8 +122,38 @@ struct SnapRealm {
...
+  unsigned num_open_past_parents;
```
Comment (joscollin) [inline src/mds/SnapRealm.h]: private

**Comment by varshar16** [comment]
@joscollin Why do you want to move all the data members to private?

**Comment by joscollin** [comment]
> @joscollin Why do you want to move the data members to private?

You need to do some rework and figure that out by yourself.

**Comment by varshar16** [comment]
> > @joscollin Why do you want to move the data members to private?
> 
> You need to do some rework and figure that out by yourself.

Thank you for responding. I will check it.

**Pull request merged**